src/data/ons_pd.py

```diff
@@ -29,8 +29,12 @@ def __init__(self, ons_pd_csv_path, load_data=True, index_column=None, fields=No
 
         if load_data:
             self.logger.debug(f"Loading ONS data from {ons_pd_csv_path} with the following data:\n{self.fields}")
-            self.data = pd.read_csv(ons_pd_csv_path, index_col=index_column, usecols=self.fields, dtype=data_types, encoding="utf-8")
 
+            # Handle index column possibly not existing
+            try:
+                self.data = pd.read_csv(ons_pd_csv_path, index_col=index_column, usecols=lambda col: col in self.fields, dtype=data_types, encoding="utf-8")
+            except ValueError:
```

Comment:
Do you think it would be sensible to add a logger.error line in here?

Reply:
This is deliberate - the file either has an index column (pcd) if the full file is being used, or if the reduced file is being used there is no index column.

Fairly sure this is the most pythonic way of doing it but will add some comments and logging

Reply:
Have now added commit to hopefully address this

Reply:
Ah yes that's good :)